### PR TITLE
Fix editorconfig action

### PR DESCRIPTION
**Pull request by figsoda**
###### Description of changes

**Draft because I need to revert the last 2 commits. This is ready for review**

second attempt ([previous](https://github.com/NixOS/nixpkgs/pull/218850)) at fixing this, also updates editorconfig-checker to 2.7.0

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by drupol** [comment]
Hey,

Just chiming here... we also had issue with that action lately (see https://github.com/fossar/nix-phps/pull/211)

Idea: How about switching to [`prettier`](https://prettier.io/) instead of the Github Action (example: https://github.com/loophp/collection/blob/master/.github/workflows/prettier.yml) WDYT ?

**Comment by figsoda** [comment]
it probably had something to do with `nix-env` instead, I'm not sure if this breakage is enough to warrant a change in the tool we are using

File: pkgs/tools/nix/nix-init/default.nix [hunk 1]
```diff
@@ -12,7 +12,7 @@
 , callPackage
 , spdx-license-list-data
 }:
-
+
```
Comment (mweinelt) [inline pkgs/tools/nix/nix-init/default.nix:15]:
```suggestion

```

Comment (figsoda) [reply pkgs/tools/nix/nix-init/default.nix:15]: The last 2 commits are there to test the action, I can revert them once this is approved

**Comment by cole-h** [comment]
I think this is the "right" solution for now: https://github.com/NixOS/nixpkgs/pull/218858

We can chase the actual issue upstream (namely, `~/.nix-profile/bin` not existing in `$PATH`).

**Pull request closed without merging**